src/query/query_manager.c

```diff
@@ -2841,7 +2786,7 @@ qmgr_free_temp_file_list (THREAD_ENTRY * thread_p, QMGR_TEMP_FILE * tfile_vfid_p
       fd_ret = NO_ERROR;
       if ((tfile_vfid_p->temp_file_type != FILE_QUERY_AREA || is_error) && !VFID_ISNULL (&tfile_vfid_p->temp_vfid))
 	{
-	  if (was_preserved)
+	  if (was_preserved || tfile_vfid_p->temp_file_type == FILE_QUERY_AREA)
```

Comment:
This if condition seems not clear.
there exists "tfile_vfid_p->temp_file_type != FILE_QUERY_AREA" at outer if-statement and tfile_vfid_p->temp_file_type == FILE_QUERY_AREA at inner if-statement.

Reply:
have to check the error case (is_error is true)

Reply:
I fixed it as using preserved flag in QMGR temp_file.